### PR TITLE
MP-1631 Active contract filter for pickup locations.

### PR DESCRIPTION
**Pull request by NickVries**
**Changes**
- Added the `onlyActiveContracts` parameter to getPickupDropoffLocations method.
- The `onlyActiveContracts` parameter defaults to `true`.

**Related Issues**
- https://myparcelcombv.atlassian.net/browse/MP-1631

### CODE REVIEW

File: src/MyParcelComApi.php [hunk 1]
```diff
@@ -837,17 +839,38 @@ private function arrayToFilter(array $array)
...
 
         return $filters;
     }
+
+    /**
+     * @param bool                  $onlyActiveContracts
+     * @param null|CarrierInterface $specificCarrier
+     * @return array
+     */
+    private function determineCarriers($onlyActiveContracts, $specificCarrier = null)
```
Comment (denniskoster) [inline src/MyParcelComApi.php]: Missing typehints and return type

File: src/MyParcelComApi.php [hunk 2]
```diff
@@ -837,17 +839,38 @@ private function arrayToFilter(array $array)
...
-                $filters['filter[has_contract]'] = $value;
-            }
-
             $filters["filter[$name]"] = $value;
         }
 
         return $filters;
     }
+
+    /**
```
Comment (denniskoster) [inline src/MyParcelComApi.php]: I think some sort of explanation on what does method does would be welcome. Specifically the `$specificCarrier` parameter isn't really clear to me.

File: src/MyParcelComApi.php [hunk 3]
```diff
@@ -837,17 +839,38 @@ private function arrayToFilter(array $array)
...
+     * @param bool                  $onlyActiveContracts
+     * @param null|CarrierInterface $specificCarrier
+     * @return array
+     */
+    private function determineCarriers($onlyActiveContracts, $specificCarrier = null)
+    {
```
Comment (denniskoster) [inline src/MyParcelComApi.php]: I think an early return would be possible here.
```
if ($onlyActiveContracts === false && $specificCarrier !== null) {
    return $specificCarrier;
}
```

File: src/MyParcelComApiInterface.php [hunk 4]
```diff
@@ -57,21 +57,23 @@ public function getCarriers();
...
      * Get the pick up/drop off locations around a given location. If no carrier
-     * is given, the default carrier is used.
+     * is given, pick up locations for all available carriers will be used.
      *
      * @param string                $countryCode
      * @param string                $postalCode
      * @param string|null           $streetName
      * @param string|null           $streetNumber
      * @param CarrierInterface|null $specificCarrier
+     * @param                       $onlyActiveContracts
```
Comment (denniskoster) [inline src/MyParcelComApiInterface.php]:
```
@param bool $onlyActiveContract
```

File: src/MyParcelComApi.php (hunk 1 above)
Comment (Shodske) [reply src/MyParcelComApi.php]: The SDK supports PHP 5.6. So there is no support for return types or scalar type hints.

File: src/MyParcelComApi.php [hunk 5]
```diff
@@ -837,17 +839,50 @@ private function arrayToFilter(array $array)
...
+        // If we're looking for a specific carrier but it doesn't
+        // matter if it has active contracts, just return it immediately.
+        if (!$onlyActiveContracts && $specificCarrier) {
+            return [$specificCarrier];
+        }
+
+        // Return all carriers if we're not filtering for anything
+        // specific.
+        if (!$onlyActiveContracts && !$specificCarrier) {
```
Comment (Shodske) [inline src/MyParcelComApi.php]: You can reduce this check to just `if (!$onlyActiveContracts)` as the previous check already ensures that `$specificCarrier` will be falsy if `$onlyActiveContract` is falsy.

**Pull request merged**